File: app/src/main/java/era/apps/happinessjar/util/database/AppMessage.java
package era.apps.happinessjar.util.database;


import android.animation.ObjectAnimator;
import android.view.View;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "Message") // room table name  for Message
public   class AppMessage {
    // create id  autoGenerate and PrimaryKey for the single message to catch it
    @PrimaryKey(autoGenerate = true)
    private int id;

    private String content;    // the content of the single Message
    private int status;        // the status of the single message is Liked or No
    private String type;       // we have 2 type of message whatsApp Message and the App Message

    public AppMessage(int status, String content, String type) {
        this.content = content;
        this.status = status;
        this.type = type;



           }

    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public String getContent() {
        return content;
    }

    public int getStatus() {
        return status;
    }

    public String getType() {
        return type;
    }

    public boolean isLiked() {
        return status == 1;
    }
}

File: app/src/main/java/era/apps/happinessjar/data/models/AppMessage.kt
package era.apps.happinessjar.data.models

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity(tableName = "Message") // room table name  for Message
// room table name  for Message
data class AppMessage(
        var status: Int,
        var content: String,
        val type: String,
){
    @PrimaryKey(autoGenerate = true) var id: Int=0


    @Ignore
    fun isLiked(): Boolean {
        return status == 1
    }

}